monai/engines/evaluator.py

```diff
@@ -295,14 +295,16 @@ def _iteration(self, engine: SupervisedEvaluator, batchdata: dict[str, torch.Ten
 
         # put iteration outputs into engine.state
         engine.state.output = {Keys.IMAGE: inputs, Keys.LABEL: targets}
-
-        # execute forward computation
-        with engine.mode(engine.network):
-            if engine.amp:
-                with torch.cuda.amp.autocast(**engine.amp_kwargs):
+        if hasattr(engine.network, "training"):
```

Comment:
Suggest to add the reference (this line: https://github.com/pytorch/pytorch/blob/4a1b9717481070e204c9d2049213c24457ba8592/torch/jit/_script.py#L594) otherwise users may feel confused why "training" here is needed in an evaluator.